pkg/android/get-android-info.go

```diff
@@ -0,0 +1,238 @@
+}
+
+type AndroidManifestApplicationData struct {
+	XMLName  xml.Name                `xml:"application"`
+	MetaData AndroidManifestMetaData `xml:"meta-data"`
+}
+
+type AndroidManifestMetaData struct {
+	XMLName xml.Name `xml:"meta-data"`
+	Name    []string `xml:"name,attr"`
+	Value   []string `xml:"value,attr"`
+}
+
+func isXMLContent(buffer []byte) bool {
+	xmlHeader := []byte("<manifest xmlns")
+	return bytes.Contains(buffer, xmlHeader)
+}
+
+func isProtobufContent(path string) (bool, error) {
+	// Open the file
+	file, err := os.Open(path)
+	if err != nil {
+		return false, err
+	}
+	defer file.Close()
+
+	// Read the first 8 bytes of the file
+	buffer := make([]byte, 8)
+	_, err = file.Read(buffer)
+	if err != nil {
+		return false, err
+	}
+
+	// Check for the protobuf identifier
+	protobufIdentifier := []byte{0x0A} // Example: Protobuf identifier often starts with 0x0A
+	return bytes.HasPrefix(buffer, protobufIdentifier), nil
+}
+
+// Function to extract the android manifest file from a given aab file
+func GetAndroidManifestFileFromAAB(path string) (string, error) {
+	aabManifestPath := filepath.Join("base", "manifest", "AndroidManifest.xml")
+	outputPath := filepath.Join(path, "..", filepath.Base(aabManifestPath))
+
+	if filepath.Ext(path) == ".aab" {
+		zipData, err := zip.OpenReader(path)
+
+		if err != nil {
+			return "", err
+		}
+
+		defer zipData.Close()
+
+		for _, file := range zipData.File {
+			if file.Name == aabManifestPath {
+				zippedFile, err := file.Open()
+				if err != nil {
+					return "", nil
+				}
+				defer zippedFile.Close()
+
+				destinationFile, err := os.Create(outputPath)
+
+				if err != nil {
+					return "", err
+				}
+
+				defer destinationFile.Close()
+
+				_, err = io.Copy(destinationFile, zippedFile)
+				if err != nil {
+					return "", err
+				}
+
+				return outputPath, nil
+			}
+		}
+	}
+	return "", fmt.Errorf("no aab file")
+}
+
+// GetAndroidXMLData - Pulls information from a human-readable xml file into a struct
+func GetAndroidXMLData(manifestFile string) (*AndroidManifestData, error) {
+	data, err := os.ReadFile(manifestFile)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read " + manifestFile + " : " + err.Error())
+	}
+
+	var manifestData *AndroidManifestData
+	err = xml.Unmarshal(data, &manifestData)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse data from " + manifestFile + " : " + err.Error())
+	}
+
+	return manifestData, nil
+}
+
+func GetAndroidProtobufData(path string) (*AndroidManifestData, error) {
+	aabManifestData := make(map[string]string)
+
+	content, err := os.ReadFile(path)
+
+	if err != nil {
+		return nil, err
+	}
+
+	rawAabManifestData := &proto_messages.XmlNode{}
+
+	err = proto.Unmarshal(content, rawAabManifestData)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, data := range rawAabManifestData.GetElement().GetAttribute() {
+		if data.ResourceId == 16843291 || data.Name == "versionCode" {
```

Comment:
It might be worth having a comment to document where these numbers come from:
https://developer.android.com/reference/android/R.attr#versionCode

Or better yet: move them to a constant and put the comment there?